main.py

```diff
@@ -263,25 +254,42 @@ def _kill_tb_child():
     atexit.register(_kill_tb_child)
 
 
-def get_best_checkpoint_path(run_dir, phase, task_name=None):
+def get_best_checkpoint_path(args, phase, task_name=None):
     """ Look in run_dir for model checkpoint to load.
     Hierarchy is
-        1) best task-specific checkpoint for target_train, used when evaluating
+        1) user-specified target task checkpoint
+        2) best task-specific checkpoint for target_train, used when evaluating
         2) if we do only target training without pretraining, then load checkpoint before
         target training
         3) if we're doing pretraining, then load the overall best model state
-        4) nothing found (empty string)
+        4) user-specified eval checkpoint
+        5) None
+
     """
     checkpoint = []
-    if phase == "target_train":
-        assert task_name is not None, "Specify a task checkpoint to evaluate from."
-        checkpoint = glob.glob(
-            os.path.join(run_dir, task_name, "model_state_%s_epoch_*.best.th" % phase)
+    if phase == "target_train" and args.load_target_train_checkpoint not in ("none", ""):
+        checkpoint = glob.glob(args.load_target_train_checkpoint)
+        assert len(checkpoint) > 0, (
+            "Specified load_target_train_checkpoint not found: %r"
+            % args.load_target_train_checkpoint
         )
-    if len(checkpoint) == 0:
-        checkpoint = glob.glob(os.path.join(run_dir, "model_state_untrained_pre_target_train.th"))
-    if len(checkpoint) == 0 and phase == "pretrain":
-        checkpoint = glob.glob(os.path.join(run_dir, "model_state_pretrain_epoch_*.best.th"))
+
+    if phase == "pretrain":
```

Comment:
Why this condition? The simplest case (the one in demo.py) is to do both pretraining and fine-tuning, and if we're doing that, we should be loading this checkpoint when starting fine-tuning.

Reply:
Good point- i'll put this under target_train phase instead